### PR TITLE
Reader: url should be an NSURL not a String.

### LINKED ISSUE #4402: Reader: Inconsistent sharing options in Reader list and full post views
In the Reader, there are two options sharing a post:
1. In the Reader list view, you can tap the 3 dots in the top right and select "Share"
2. In the full post view, you can tap the sharing icon in the top right

The resulting sharing options are not consistent. Most notable is that you can't share the post to WordPress unless you're in the full post view. Is that intentional? It can be confusing if you're looking for a way to share a post e.g. on your own blog and don't notice that the full post view gives you more sharing options.

Sharing options in Reader list view:

Sharing options in full post view:

For comparison, the web Reader gives you the same three sharing options (WordPress, Twitter, Facebook) in both the list view and the full post view:

screen shot 2015-11-03 at 4 55 57 pm

Viewed on iPhone 6, iOS 9.1, WPiOS beta 5.7.0.20151102

### PR DESCRIPTION
**Pull request by aerych**
Closes #4402 
The share link was being set as a String instead of an NSURL. This was causing our application share targets to not appear. 

To test: 
Go to the reader post list, tap the ( ... ) icon in one of the post cards and choose "Share" in the action sheet.  Confirm that WordPress, etc. are included as share options. 

Needs Review: @astralbodies 

@astralbodies The optional unboxing in this PR works but I think its a bit ugly.  I'd be game to go a different route but couldn't think of one I liked better. Suggestions welcome!

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift [hunk 1]
```diff
@@ -17,8 +17,10 @@ public class ReaderHelpers {
         }
 
         activityItems.append(postDictionary)
-        if let url = link {
-            activityItems.append(url)
+        if let urlPath = link {
+            if let url = NSURL(string: urlPath) {
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift]: You could concatenate the ifs -
```
if let urlPath = link, url = NSURL(string: urlPath) {
```

**Comment by astralbodies** [comment]
:shipit: Even if you don't concatenate the if statements. :smile:

File: WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift (hunk 1 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift]: :+1:

**Comment by aerych** [comment]
Thanks @astralbodies

**Pull request merged**